brainclouds2s.js

```diff
@@ -241,31 +188,44 @@ function request(context, json, callback)
             return
         }
 
-        context.requestQueue.splice(0, 1); // Remove this request
+        if (data && data.messageResponses && data.messageResponses.length > 0) {
+            callCallback(callback, context, data.messageResponses[0])
+        }
+        else {
+            
+            // Error that has no packets
+            callCallback(callback, context, data)
+        }
+
+        // This request is complete and safe to remove from queue after checking for callback
```

Comment:
Minor: I'd change "after checking for callback" to "now that callback has been invoked"